server/app/views/admin/questions/QuestionsListView.java

```diff
@@ -343,51 +349,79 @@ private Optional<ButtonTag> renderQuestionTranslationLink(QuestionDefinition def
-    return asRedirectElement(
-        makeSvgTextButton(linkText, Icons.VISIBILITY)
-            .withClasses(AdminStyles.TERTIARY_BUTTON_STYLES),
-        link);
-  }
-
   private Pair<TdTag, Optional<Modal>> renderActionsCell(
       Optional<QuestionDefinition> active,
       Optional<QuestionDefinition> draft,
       ActiveAndDraftQuestions activeAndDraftQuestions,
       Http.Request request) {
-    TdTag td = td().withClasses(BaseStyles.TABLE_CELL_STYLES);
+    TdTag td =
+        td().withClasses(
+                BaseStyles.TABLE_CELL_STYLES,
+                Styles.FLEX,
+                Styles.SPACE_X_2,
+                Styles.PR_6,
+                Styles.FONT_MEDIUM);
+    ImmutableList.Builder<DomContent> extraActions = ImmutableList.builder();
+    ButtonTag editButton = null;
     if (active.isPresent()) {
       if (draft.isEmpty()) {
         // Active without a draft.
-        td.with(renderQuestionViewLink(active.get(), "View"));
-        td.with(renderQuestionEditLink(active.get(), "New Version"));
+        editButton = renderQuestionEditLink(active.get(), "New Version");
       } else if (draft.isPresent()) {
         // Active with a draft.
-        td.with(renderQuestionViewLink(active.get(), "View Published"));
-        td.with(renderQuestionEditLink(draft.get(), "Edit Draft"));
+        editButton = renderQuestionEditLink(draft.get(), "Edit Draft");
         Optional<ButtonTag> maybeTranslationLink = renderQuestionTranslationLink(draft.get());
         if (maybeTranslationLink.isPresent()) {
-          td.with(maybeTranslationLink.get());
+          extraActions.add(maybeTranslationLink.get());
         }
-        td.with(renderDiscardDraftLink(draft.get(), request));
+        extraActions.add(renderDiscardDraftLink(draft.get(), request));
       }
     } else if (draft.isPresent()) {
       // First revision of a question.
-      td.with(renderQuestionEditLink(draft.get(), "Edit Draft"));
       Optional<ButtonTag> maybeTranslationLink = renderQuestionTranslationLink(draft.get());
+      editButton = renderQuestionEditLink(draft.get(), "Edit Draft");
       if (maybeTranslationLink.isPresent()) {
-        td.with(maybeTranslationLink.get());
+        extraActions.add(maybeTranslationLink.get());
       }
     }
     // Add Archive options.
     QuestionDefinition questionForArchive = draft.isPresent() ? draft.get() : active.get();
     Pair<DomContent, Optional<Modal>> archiveOptionsAndModal =
         renderArchiveOptions(questionForArchive, activeAndDraftQuestions, request);
-    DomContent archiveOptions = archiveOptionsAndModal.getLeft();
-    Optional<Modal> maybeArchiveModal = archiveOptionsAndModal.getRight();
-    td.with(archiveOptions);
-    return Pair.of(td, maybeArchiveModal);
+    extraActions.add(archiveOptionsAndModal.getLeft());
+
+    // Build extra actions button and menu.
+    String extraActionsButtonId = "extra-actions-" + questionForArchive;
```

Comment:
Since the actual ID needn't be human readable, what about using `Modal.randomModalId` to generate a unique identifier? Side note is that we could probably pull the "random ID" function up to something higher level than just modal.

Reply:
Done.